### ISSUE TITLE
ChatMessage doesn't get appended in the ChatContext

### ISSUE THREAD

**Issue by naman-scogo** [livekit/agents#408]
I have made a Voice Assistant using the [kitt example](https://github.com/livekit/agents/blob/main/examples/_deployed/kitt/kitt.py)

It can interact over voice as well as text messages.

Once the user ends the call/room, the assistant is designed to process all the messages and store them in a persistent database.

However, after the call ends, all the ChatMessaged pushed to the ChatContext using llm.chat() method while answering the text messages from the user are not present in the messages array.

The same happens if I push a System Message to the context while sending a system message to the llm after the function calls are finished.

**Comment by Reynold97** [comment]
Hi, friend. I was trying to run the same example but I can't. On my windows machine I always have this error:

{"message": "silero stream failed\nTraceback (most recent call last):\n  File \"D:\\!!!Trabajo\\!!Natasquad\\Repos\\Live-Agent\\env\\lib\\site-packages\\livekit\\plugins\\silero\\vad.py\", line 148, in _run\n    await asyncio.shield(self._run_inference())\n  File \"D:\\!!!Trabajo\\!!Natasquad\\Repos\\Live-Agent\\env\\lib\\site-packages\\livekit\\plugins\\silero\\vad.py\", line 171, in _run_inference\n    raw_prob = await asyncio.to_thread(\n  File \"D:\\!!!Trabajo\\!!Natasquad\\Repos\\Live-Agent\\env\\lib\\asyncio\\threads.py\", line 25, in to_thread\n    return await loop.run_in_executor(None, func_call)\n  File \"D:\\!!!Trabajo\\!!Natasquad\\Repos\\Live-Agent\\env\\lib\\concurrent\\futures\\thread.py\", line 58, in run\n    result = self.fn(*self.args, **self.kwargs)\n  File \"D:\\!!!Trabajo\\!!Natasquad\\Repos\\Live-Agent\\env\\lib\\site-packages\\livekit\\plugins\\silero\\vad.py\", line 172, in <lambda>\n    lambda: self._model(tensor, self._sample_rate).item()\n  File \"C:\\Users\\Reynold/.cache\\torch\\hub\\snakers4_silero-vad_v4.0\\utils_vad.py\", line 65, in __call__\n    ort_outs = self.session.run(None, ort_inputs)\n  File \"D:\\!!!Trabajo\\!!Natasquad\\Repos\\Live-Agent\\env\\lib\\site-packages\\onnxruntime\\capi\\onnxruntime_inference_collection.py\", line 220, in run\n    return self._sess.run(output_names, input_feed, run_options)\nonnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Unexpected input data type. Actual: (tensor(int32)) , expected: (tensor(int64))\n", "job_id": "AJ_juzTLFapA3Pq", "pid": 12844, "timestamp": "2024-07-02T20:29:42.814037+00:00"}

TL:DR

[ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Unexpected input data type. Actual: (tensor(int32)), expected: (tensor(int64))

**Comment by cmurrayis** [comment]
Hi,

I seem to be getting something similar:

```
unhandled exception in the job entry <function entrypoint at 0x7fa47b98c4a0> | exc_info=Traceback (most recent call last):
  File "/home/appuser/myagent.py", line 13, in entrypoint
    initial_ctx = ChatContext().append(
                  ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ChatContext' object has no attribute 'append' timestamp=2024-07-11T05:38:51.319684+00:00 
unhandled exception in the job entry <function entrypoint at 0x7fa47b98c4a0>
Traceback (most recent call last):
  File "/home/appuser/myagent.py", line 13, in entrypoint
    initial_ctx = ChatContext().append(
                  ^^^^^^^^^^^^^^^^^^^^
```

Which is just running the minimal-agent.py example. When I looked further into the livekit.agents.llm module there does not appear to be an "append" attribute outlined:

```
class ChatContext(builtins.object)
 |  ChatContext(messages: 'list[ChatMessage]' = <factory>) -> None
 |
 |  ChatContext(messages: 'list[ChatMessage]' = <factory>)
 |
 |  Methods defined here:
 |
 |  eq(self, other)
 |      Return self==value.
 |
 |  init(self, messages: 'list[ChatMessage]' = <factory>) -> None
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  repr(self)
 |      Return repr(self).
 |
 |  copy(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors defined here:
 |
 |  dict
 |      dictionary for instance variables (if defined)
 |
 |  weakref
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  annotations = {'messages': 'list[ChatMessage]'}
 |
 |  dataclass_fields = {'messages': Field(name='messages',type='list[C...
 |
 |  dataclass_params = _DataclassParams(init=True,repr=True,eq=True,or...
 |
 |  hash = None
 |
 |  match_args = ('messages',)
```

**Comment by theomonnom** [comment]
Hey, it seems you were trying our updated examples without using our latest agents version. Please try again using livekit-agents v0.8.0
